### ISSUE TITLE
No browser refresh when changing HTML resources

### ISSUE THREAD

**Issue by julienguiraud** [BrowserSync/grunt-browser-sync#120]
Hello,

After installing a new app with JHipster, running grunt serve and changing `'src/main/webapp/scripts/app/entities/talk/talk-detail.html'` file, the browser does not refresh.

Workaround:
When I insert the path `'src/main/webapp/scripts/**/*.html` in browserSync.dev.bsFiles.src like above, that works but only after relaunching `grunt serve`.

```javascript
browserSync: {
            dev: {
                bsFiles: {
                    src : [
                        'src/main/webapp/**/*.html',
                        'src/main/webapp/**/*.json',
                        'src/main/webapp/assets/styles/**/*.css',
                        'src/main/webapp/scripts/**/*.js',
                        'src/main/webapp/scripts/**/*.html',                                        
                        'src/main/webapp/assets/images/**/*.{png,jpg,jpeg,gif,webp,svg}',
                        'tmp/**/*.{css,js}'
                    ]
                }
            },
            options: {
                watchTask: true,
                proxy: "localhost:8080"
            }
        },
```

Windows 7 64 bits
JHipster : 2.19.0
Chrome, FF last versions
node : 0.12.6
npm : 2.11.2
grunt-browser-sync : 2.1.2

Issues raised in 4 differents PC with almost same configuration.
[JHipster related issue] (https://github.com/jhipster/generator-jhipster/issues/1900)

Thanks

**Comment by aolin480** [comment]
I am going to try this one on you too, to see if it's the same issue I am having. lol Can you try, instead of using 'src/main/webapp/**/*.html', under bsFiles, use 'src/main/webapp/index.html', or whatever .html file you want to watch? I have the issue where browser-sync is not recognizing wildcards, but specific files. Just for funsies

**Comment by julienguiraud** [comment]
Adding `'src/main/webapp/scripts/**/*.html'`                                      
Pointing the real html file w/o wildcard works.

This above config does not work.
```javascript
browserSync: {
            dev: {
                bsFiles: {
                    src : [
                        'src/main/webapp/**/*.html',
                        'src/main/webapp/**/*.json',
                        'src/main/webapp/assets/styles/**/*.css',
                        'src/main/webapp/scripts/**/*.js',
                        'src/main/webapp/assets/images/**/*.{png,jpg,jpeg,gif,webp,svg}',
                        'tmp/**/*.{css,js}'
                    ]
                }
            },
            options: {
                watchTask: true,
                proxy: "localhost:8080"
            }
        },
```
 

This above config does work !
```javascript
browserSync: {
            dev: {
                bsFiles: {
                    src : [
                        'src/main/webapp/**/*.html',
                        'src/main/webapp/scripts/**/*.html',                                        
                        'src/main/webapp/**/*.json',
                        'src/main/webapp/assets/styles/**/*.css',
                        'src/main/webapp/scripts/**/*.js',
                        'src/main/webapp/assets/images/**/*.{png,jpg,jpeg,gif,webp,svg}',
                        'tmp/**/*.{css,js}'
                    ]
                }
            },
            options: {
                watchTask: true,
                proxy: "localhost:8080"
            }
        },
```
 So wildcard seem to work.

**Comment by chrisruddick** [comment]
I'm seeing the same issue.  Adding the wildcard line:
```
'src/main/webapp/scripts/**/*.html',
```

and relaunching `grunt serve` allowed browserSync to refresh html changes.  Prior to this, no html changes were detected within the `webapps` subdirectories.

**Comment by jdubois** [comment]
I'm the [JHipster](http://jhipster.github.io/) lead dev and I confirm we have this issue.

The solution with Chrome is to disable the cache when the dev tools is opened, but previously it was working without doing this, so there is a regression here. And that's a really bad first impression for users.

**Comment by shakyShane** [comment]
@jdubois - fixing shortly with a patch release

**Comment by shakyShane** [comment]
All: 

To receive the upstream fixes from browser-sync-client, please run the following command.

`npm rm grunt-browser-sync && npm i grunt-browser-sync`

Alternatively:

```shell
cd node_modules/grunt-browser-sync/node_modules/browser-sync
npm install browser-sync-client@latest
```

**Comment by julienguiraud** [comment]
I just updated to `"grunt-browser-sync": "^2.1.3"` and that does not work.

When adding, as before, `'src/main/webapp/scripts/**/*.html'`in `src` property in `Gruntfile.js`, that works !

```
bsFiles: {
    src : [
        'src/main/webapp/**/*.html',
        'src/main/webapp/scripts/**/*.html',                        
        'src/main/webapp/**/*.json',
        'src/main/webapp/assets/styles/**/*.css',
        'src/main/webapp/scripts/**/*.js',
        'src/main/webapp/assets/images/**/*.{png,jpg,jpeg,gif,webp,svg}',
        'tmp/**/*.{css,js}'
        ]
}
```

@jdubois : Note that being JHipster leader is very helpful to get quickly a fix ! :wink:

**Comment by shakyShane** [comment]
Can you run 'npm ls browser-sync' please

**Comment by julienguiraud** [comment]
here it is.

![capture_npm ls browser-sync](https://cloud.githubusercontent.com/assets/6605556/9671446/3e0d3198-5293-11e5-9111-71a3d4fbe91e.png)

**Comment by jahu** [comment]
i have the same problem with browser-sync 2.9.3:

![image](https://cloud.githubusercontent.com/assets/7642643/9933946/bdfa5708-5d4e-11e5-8aa5-0f669b58c724.png)

**Comment by julienguiraud** [comment]
Could we reopen this issue?

**Comment by jdubois** [comment]
@julienguiraud do you have the issue with the 2.21.0 version? In that case that could be another issue, as this was supposed to be corrected, so you can open another one.
Please provide also your OS, as it might be OS-related.

**Comment by aolin480** [comment]
@julienguiraud I would try a different approach and just use regular browser-sync in your gruntfile. I had so many issues on Windows with grunt-browser-sync after 2.6.0, and I followed the @shakyShane tutorial, and I can now use the most up-to-date browser-sync without even using the grunt-browser-sync grunt module. Try it out and see if it works for you. grunt-browser-sync is just a wrapper for browser-sync anyway. Try it straight from the source first. http://www.shakyshane.com/javascript/nodejs/browser-sync/2014/08/24/browser-sync-plus-grunt/. One thing I want to point out when trying out just regular browser-sync is to make sure you have the following in your watch task. spawn: false. I easily skipped it, and it didn't work, but as soon as I added it, I was golden.

```
watch: {
    options: {
       spawn: false
    },
   // your other tasks
}
```

```
grunt install browser-sync -g
```

**Comment by sisu4u** [comment]
I'm having the same issue with a jhipster app. I'm running on browser-sync 2.9.3.

Adding the line as @julienguiraud mentions helps.

One thing I also notice is that every time I save a file it shows "File changed" twice in the command window. This is, however, independent if the line is added or not and might be also in issue related to jhipster.
